synapse/storage/databases/main/room.py

```diff
@@ -409,22 +409,32 @@ async def get_rooms_paginate(
             limit: maximum amount of rooms to retrieve
             order_by: the sort order of the returned list
             reverse_order: whether to reverse the room list
-            search_term: a string to filter room names by
+            search_term: a string to filter room names,
+                canonical alias and room ids by
         Returns:
             A list of room dicts and an integer representing the total number of
             rooms that exist given this query
         """
         # Filter room names by a string
         where_statement = ""
+        search_pattern = ""
```

Comment:
Let's make `search_pattern` a list throughout, rather than a string sometimes and a list others:

```suggestion
        search_pattern = []
```